test/index.js

```diff
@@ -789,6 +789,94 @@ describe('limit tests', () => {
     assert(result.events)
   })
 
+  it('should set as invalid circular property dependency', () => {
+    const waf = new DDWAF(processor)
+    const context = waf.createContext()
+    const payload = {
+      key: 'value',
+      mail: 'from@domain.com'
+    }
+    payload.child1 = payload
+    payload.child2 = payload
+    payload.child3 = payload
+
+    const result = context.run({
+      persistent: {
+        'server.request.body': payload,
+        'waf.context.processor': { 'extract-schema': true }
+      }
+    }, TIMEOUT)
+
+    assert.deepStrictEqual(result.derivatives, {
+      'server.request.body.schema': [
+        {
+          mail: [8],
+          key: [8],
+          child1: [0],
+          child2: [0],
+          child3: [0]
+        }
+      ]
+    })
+  })
+
+  it('should iterate same instances in array', () => {
+    const waf = new DDWAF(processor)
+    const context = waf.createContext()
+    const item = {
+      key: 'value',
+      mail: 'from@domain.com'
+    }
+
+    const payload = []
+    payload.push(item)
+    payload.push(item)
+    payload.push(item)
+    payload.push(item)
+
+    const result = context.run({
+      persistent: {
+        'server.request.body': payload,
+        'waf.context.processor': { 'extract-schema': true }
+      }
+    },
+    TIMEOUT)
+
+    assert(result.derivatives['server.request.body.schema'][1].len === 4)
```

Comment:
that assert is not enough right ? the length could be 4 but it could be 4 invalids as far as I can see

Reply:
True, checking whoe payload now